[JIT] [Easy] Factor out non tensor peephole

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56966 [JIT] Disable Complete Shape Inlining For Testing Purposes
* #56828 Add unary/binary ops necessary for mobilenet
* #56642 Only optimize while change has been made
* #56438 [JIT] integer value refinement
* #56437 Factor out isDominatedBy
* #55978 Add Value * == Value * peephole
* **#55977 [JIT] [Easy] Factor out non tensor peephole**
* #55926 Add list len refinement
* #55925 Add handling of symbolic shapes
* #54809 Initial Symbolic Shape Analysis

Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes.

Commit:
Factor out non tensor peephole

[ghstack-poisoned]

Krovatkin commented on torch/csrc/jit/passes/peephole_non_tensor.cpp:
```diff
@@ -0,0 +1,142 @@
+        }
+      }
+
+      // [aliasing sensitive optimizations]
```
nitpick: is this comment still relevant?

Krovatkin reviewed (approved):
:shipit:

penguinwu reviewed (approved):
LGTM

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

penguinwu reviewed (approved):
Did not look very closely, but LGTM

desertfire commented on torch/csrc/jit/passes/peephole_non_tensor.cpp:
```diff
@@ -0,0 +1,134 @@
+
+bool PeepholeOptimizeNonTensor(const std::shared_ptr<Graph>& graph) {
+  PeepholeOptimizeNonTensorImpl peephole(graph);
+  bool changed = peephole.run();
```
Do we need to apply EliminateDeadCode here like in https://github.com/pytorch/pytorch/blob/72c3ee073f982c69d532435972a16bd9958413bd/torch/csrc/jit/passes/peephole.cpp#L551 ?

eellison replied:
This is being invoked as part of PeepholeOptimize which has it's own DCE call so i'm not invoking it here

desertfire reviewed (commented):
LGTM

Commit:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).

eellison committed:
Update on "[JIT] [Easy] Factor out non tensor peephole"


Differential Revision: [D27755329](https://our.internmc.facebook.com/intern/diff/D27755329)

Purely a mechanical change, no code changes. 

[ghstack-poisoned]

eellison commented:
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27755329).
